test/fast/Distributions.test.ts

```diff
@@ -1,22 +1,176 @@
 import * as chai from "chai";
+import { expect } from "chai";
 import { solidity } from "ethereum-waffle";
-import { smock } from "@defi-wonderland/smock";
+import { BigNumber } from "ethers";
+import { FakeContract, MockContract, smock } from "@defi-wonderland/smock";
+import { deployments, ethers } from "hardhat";
+import { SignerWithAddress } from "hardhat-deploy-ethers/signers";
+import { Issuer, Marketplace, Distribution, Distribution__factory, IERC20, Fast, FastTokenFacet } from "../../typechain";
+import { abiStructToObj, DistributionPhase } from "../utils";
+import { fastFixtureFunc } from "../fixtures/fast";
 
 chai.use(solidity);
 chai.use(smock.matchers);
 
-describe("Distributions", () => {
-  before(async () => { });
-  beforeEach(async () => { });
+describe.only("Distributions", () => {
+  let deployer: SignerWithAddress,
+    issuerMember: SignerWithAddress,
+    governor: SignerWithAddress,
+    alice: SignerWithAddress;
+
+  let issuer: FakeContract<Issuer>,
+    marketplace: FakeContract<Marketplace>,
+    fast: Fast,
+    token: FastTokenFacet,
+    erc20: FakeContract<IERC20>;
+
+  const fastDeployFixture = deployments.createFixture(fastFixtureFunc);
+
+  before(async () => {
+    // Keep track of a few signers.
+    [deployer, issuerMember, governor, alice] = await ethers.getSigners();
+
+    // Mock an Issuer and an Marketplace contract.
+    issuer = await smock.fake("Issuer");
+    marketplace = await smock.fake("Marketplace");
+    marketplace.issuerAddress.returns(issuer.address);
+
+    marketplace.isMember.whenCalledWith(governor.address).returns(true);
+    marketplace.isMember.returns(false);
+    marketplace.isActiveMember.whenCalledWith(governor.address).returns(true);
+    marketplace.isActiveMember.returns(false);
+
+    // TODO: Totally forgotten about what I'm doing... I just want a generic ERC20.
+    erc20 = await smock.fake("IERC20");
```

Comment:
This should work - if you want internal accuracy (allowances etc) you can use the `contracts/dummy/ERC20` contract. Won't be ever deployed, but it's useful to test stuff from the console too.
But honestly I prefer your approach, so we don't rely on Solidity mocks for our tests.